plugins/hls-hlint-plugin/src/Ide/Plugin/Hlint.hs

```diff
@@ -125,13 +125,21 @@ import           System.Environment                                 (setEnv,
 import           Text.Regex.TDFA.Text                               ()
 -- ---------------------------------------------------------------------
 
-newtype Log
+data Log
   = LogShake Shake.Log
+  | LogApplying NormalizedFilePath (Either String WorkspaceEdit)
+  | LogGeneratedIdeas NormalizedFilePath [[Refact.Refactoring Refact.SrcSpan]]
+  | LogGetIdeas NormalizedFilePath
+  | LogUsingExtensions NormalizedFilePath [String] -- Extension is only imported conditionally, so we just stringify them
```

Comment:
I think the fields should be banged, although it probably doesn't matter since our current recorders will force the fields anyway, and there are composite types that prevent some forcing, and we probably will never buildup huge thunks of NormalizedFilePath or the composite types, and I noticed I didn't bang the wrapper constructor fields everywhere anyway.

Reply:
I just added `StrictData` to the module, which is probably a sensible default these days anyway.